lib/student.ex

```diff
@@ -16,6 +17,49 @@ defmodule Student do
     defstruct [:pid]
   end
 
+  defmodule Event do
+    use GenServer
+
+    @type t :: %__MODULE__{
+            subscribers: list(pid())
+          }
+    defstruct subscribers: []
+
+    # client side
+
+    def new() do
+      GenServer.start_link(__MODULE__, :ok)
+    end
+
+    def subscribe(student_repr, pid) do
+      {:ok, event_manager} = Student.event_manager(student_repr)
+      GenServer.call(event_manager, {:subscribe, pid})
+    end
+
+    def dispatch(student, event) do
+      GenServer.cast(student.event_manager, event)
+    end
+
+    # server side
+    @impl true
+    @spec init(any) :: {:ok, Student.Event.t()}
```

Comment:
```suggestion
    @spec init(any()) :: {:ok, Student.Event.t()}
```
Ref: https://hexdocs.pm/elixir/typespecs.html#basic-types